sr/robot/arenas/caldera_arena.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import division
+
+from math import ceil, cos, pi, sin
+
+import pygame
+
+from .arena import ARENA_FLOOR_COLOR, ARENA_MARKINGS_COLOR, ARENA_MARKINGS_WIDTH, Arena, draw_corner_zones
+from ..markers import Token
+
+
+# Perform a clockwise rotate around 0,0.
+def rotate(x, y, radians):
+    return (x * cos(radians) + y * -sin(radians)), (x * sin(radians) + y * cos(radians))
+
+
+PLATFORM_COLOUR = (0x68, 0x75, 0x91)
+
+class CalderaArena(Arena):
+    start_locations = [(-3.6, -3.6),
+                       (3.6, -3.6),
+                       (3.6, 3.6),
+                       (-3.6, 3.6)]
+
+    start_headings = [0.25 * pi,
+                      0.75 * pi,
+                      -0.75 * pi,
+                      -0.25 * pi]
+
+    def __init__(self, objects=None, wall_markers=True):
+        super(CalderaArena, self).__init__(objects, wall_markers)
+        self._init_tokens()
+
+    def _init_tokens(self):
+        token_ids = [28, 29, 30, 31]
+
+        token_locations_offsets_from_zone = [(0.5, 1.3),
+                                             (0.5, 1.5),
+                                             (0.5, 1.7),
+                                             (0.5, 1.9),
+                                             (0.5, 2.1),]
```

Comment:
There should be two tokens in each of these locations. I think it's reasonable to draw them directly on top of each other by duplicating the starting locations (assuming that this works):
``` suggestion
                                             (0.5, 2.1),] * 2
```

Reply:
it doesn't, they collide with each other and move off in random directions

Reply:
That's a shame.
We'll need to be clear in the docs for the simulator that the number of tokens is a limitation of the simulator.